src/hotWaterHandler.h

```diff
@@ -26,16 +26,14 @@ void checkHotWaterSwitch() {
             if (hotWaterSwitchReading != currStateHotWaterSwitch) {
                 currStateHotWaterSwitch = hotWaterSwitchReading;
 
-                if (currStateHotWaterSwitch == HIGH && machineState != kWaterEmpty) {
-                    if (hotWaterOn == 0) {
+                if (currStateHotWaterSwitch == HIGH) {
```

Comment:
there is no need for this change

Reply:
Sure? I think the watertank condition must be removed to be able to turn Off Hotwater If watertank becomes empty during hotvwater